src/components/api.js

```diff
@@ -0,0 +1,146 @@
+export class Api {
+	/**
+	 * param: options
+	 * Description: constructor takes a options obj that carries a bae url
+	 *                      and headers that contain auth and content-type
+	 */
+	constructor(options) {
+		this._baseUrl = options.baseUrl;
+		this._header = options.headers;
+	} //end constructor
+
+	// Helper function to handle fetch response and errors
+	_handleResponse(res) {
+		if (res.ok) {
+			return res.json();
+		} //end if
+		return Promise.reject(`Error: ${res.status}`);
+	} //end func
+
+	/****************Card routes***************/
+	// 1. GET /cards – Get all cards
+	/**
+	 * Description: gets all initial cards form url
+	 *                      json obj returned if status 200
+	 */
+	getInitialCards() {
+		return fetch(`${this._baseUrl}/cards`, {
+			method: "GET",
+			headers: this._header,
+		})
+			.then(this._handleResponse)
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._handleResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.